### PR TITLE
aws: remove, recommend `awscli` / `awscli2`

### LINKED ISSUE #176706: aws: EOL
### Describe the bug

The `aws` package isn't maintained anymore. The Nix pkg hasn't been updated in 3 years, the repo's last commit [is from over 1 year ago and states the package isn't actively developed](https://github.com/timkay/aws/commit/38ecb3a5bc6fbe8044a73d4495a4f670caf6ab14).

Most importantly, it's a source of confusion for people who actually want `awscli` or `awscli2` instead.

I believe it should be removed and replaced with an explicit `error` note that'll redirect users to `awscli`/`awscli2`.

### PR DESCRIPTION
**Pull request by kubukoz**
Closes #176706

###### Description of changes

Removes the `aws` package.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by bryanasdev000 — APPROVED** [review]
Diff and ofborg LGTM.

**Review by wamserma — APPROVED** [review]
LGTM, nixpkgs-review was also happy.

**Comment by kubukoz** [comment]
todo (self-note): add a note to the 22.11 release notes

**Comment by superherointj** [comment]
@kubukoz Please fix the conflict.

**Comment by superherointj** [comment]
@kubukoz Please fix the conflict. And report back.

**Comment by kubukoz** [comment]
@superherointj I'll update it some day this week, in general I want to tackle the changelog at the same time and I haven't found the energy to actually do it 😅

**Comment by wamserma** [comment]
ping @kubukoz

**Comment by kubukoz** [comment]
Done - sorry for the long wait. Added a note to the release notes.

**Review by bryanasdev000 — APPROVED** [review]
LGTM and ofborg is happy.

Thanks.

File: nixos/doc/manual/from_md/release-notes/rl-2211.section.xml [hunk 1]
```diff
@@ -508,6 +508,14 @@
           been removed due to lack of upstream maintenance.
         </para>
       </listitem>
+      <listitem>
+        <para>
+          The <literal>aws</literal> package has been removed due to
+          being abandoned by the upstream. It is recmomended to use
```
Comment (superherointj) [inline nixos/doc/manual/from_md/release-notes/rl-2211.section.xml:514]:
```suggestion
          being abandoned by the upstream. It is recommended to use
```

File: nixos/doc/manual/release-notes/rl-2211.section.md [hunk 2]
```diff
@@ -176,6 +176,8 @@ Available as [services.patroni](options.html#opt-services.patroni.enable).
   `python3.pkgs.influxgraph` packages, have been removed due to lack of upstream
   maintenance.
 
+- The `aws` package has been removed due to being abandoned by the upstream. It is recmomended to use `awscli` or `awscli2` instead.
```
Comment (superherointj) [inline nixos/doc/manual/release-notes/rl-2211.section.md:179]:
```suggestion
- The `aws` package has been removed due to being abandoned by the upstream. It is recommended to use `awscli` or `awscli2` instead.
```

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 176707` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>

**Comment by superherointj** [comment]
Waiting for CI, requires **Squash** merge.

**Comment by superherointj** [comment]
@ofborg eval

**Pull request merged**